server/mqtt.go

```diff
@@ -4460,17 +4611,15 @@ func isMQTTReservedSubscription(subject string) bool {
 // Common function to mqtt delivery callbacks to serialize and send the message
 // to the `cc` client.
 func (c *client) mqttEnqueuePublishMsgTo(cc *client, sub *subscription, pi uint16, qos byte, dup bool, topic, msg []byte) {
-	sw := mqttWriter{}
-	w := &sw
-
-	flags := mqttSerializePublishMsg(w, pi, qos, dup, false, topic, msg)
+	flags, headerBytes := mqttMakePublishHeader(pi, qos, dup, false, topic, len(msg))
 
 	cc.mu.Lock()
-	if sub.mqtt.prm != nil {
-		cc.queueOutbound(sub.mqtt.prm.Bytes())
-		sub.mqtt.prm = nil
+	for _, data := range sub.mqtt.prm {
```

Comment:
Although code works if `sub.mqtt.prm` is `nil`, I think having the `if` not nil and setting it to `nil` under the `if` makes more sense:
```
if sub.mqtt.prm != nil {
    for _, data := range sub.mqtt.prm {
        cc.queueOutbound(data)
    }
    sub.mqtt.prm = nil
}
```